### PR TITLE
Statistics request

### PR DESCRIPTION
**Pull request by OlegBukalov**

### CODE REVIEW

File: src/app/features/statistics/statistics.component.ts [hunk 1]
```diff
@@ -1,268 +1,139 @@
...
+      vendor: '',
+      bonus: '',
+      type: '',
+      isActive: '',
+      start: this.dateStart ?? '',
+      end: this.dateEnd ?? '',
+    };
+    this.queryParams = '';
+
+    (document.getElementById('vendor') as HTMLInputElement).value = '';
```
Comment (umilanovich) [inline src/app/features/statistics/statistics.component.ts:132]: It is very unsafe way of operating with values

File: src/app/features/statistics/statistics.component.ts [hunk 2]
```diff
@@ -1,268 +1,139 @@
...
+        .subscribe((data: IVendor) => {
+          if (data) {
+            vendorId = data.id;
+          }
+        });
+    }
+
+    this.filterParams = {
+      vendor: vendorId,
+      bonus: (document.getElementById('bonus') as HTMLInputElement).value
```
Comment (umilanovich) [inline src/app/features/statistics/statistics.component.ts:100]: Use reactive approach to change values

File: src/app/features/statistics/statistics.component.ts [hunk 3]
```diff
@@ -1,268 +1,139 @@
...
+    end: '',
+  };
 
   @ViewChild(MatPaginator) public paginator: MatPaginator;
   @ViewChild(MatSort) public sort: MatSort;
 
+  constructor(
+    public statisticsService: StatisticsService,
+    public toasterService: ToasterService,
+    public vendorsService: VendorsService,
```
Comment (rostic54) [inline src/app/features/statistics/statistics.component.ts:51]: Why all instances are public?

File: src/app/features/statistics/statistics.component.ts [hunk 4]
```diff
@@ -1,268 +1,145 @@
...
+          this.dataSource.paginator = this.paginator;
+          this.dataSource.sort = this.sort;
+        }
+      },
+      (err) => this.toasterService.showError(err, 'Some problems with getting statistics'),
+    );
+  }
+
+  public formatDates(): void {
+    this.statistics.forEach((e) => {
```
Comment (umilanovich) [inline src/app/features/statistics/statistics.component.ts:77]: `e` in callbacks usually stands for `error`. Please rename

**Review by umilanovich — APPROVED** [review]

**Pull request merged**